containers/rancher-cloud/app/run

```diff
@@ -0,0 +1,146 @@
+
+HasKubernetesNamespace()
+{       
+    kubectl get namespace "${1}" 1>/dev/null 2>/dev/null
+}
+
+CreateKubernetesNamespace()
+{
+    kubectl create namespace "${1}"
+}
+
+EnsureKubernetesNamespaceExists()
+{
+    HasKubernetesNamespace "${1}" \
+        || CreateKubernetesNamespace "${1}"
+}
+
+CreateRancherCACertificateSecrets()
+{
+	# cleanup the previously created secret so we create a fresh one every
+	# time, even though they may be the same.
+	kubectl delete secret tls-rancher -n cattle-csp-billing-adapter-system | true
+	
+	# now create a fresh one
+	kubectl get secret tls-rancher -n cattle-system -o jsonpath="{.data.tls\.crt}" | base64 -d  >> ca-additional.pem
+	kubectl create secret generic tls-ca-additional -n cattle-csp-billing-adapter-system --from-file=ca-additional.pem
+}
+
+case $action in
+    install)
+	echo "Installing Cert-Manager"
+	helm upgrade --install cert-manager  /cnab/app/charts/cert-manager --atomic --create-namespace --namespace cert-manager --set installCRDs=true ${cert_manager_image_arguments}
+	echo "Cert-Manager successfully installed"
+        echo "Installing Rancher"
+	EnsureKubernetesNamespaceExists "cattle-system"
+	helm upgrade --install rancher /cnab/app/charts/rancher/2.7.5 --atomic --namespace cattle-system --set installCRDs=true ${rancher_image_arguments} ${rancher_ingress_class_name_arguments} --set hostname=${rancher_hostname} --set bootstrapPassword=${rancher_bootstrap_password} --set replicas=${rancher_replicas} --set extraEnv[0].name=CATTLE_SERVER_URL --set extraEnv[0].value=${rancher_server_url} --set extraEnv[1].name=CATTLE_PROMETHEUS_METRICS --set-string extraEnv[1].value=true --timeout ${helm_timeout}
+        echo "Rancher successfully installed"
+        echo "Installing CSP Rancher Usage Operator CRD"
+	EnsureKubernetesNamespaceExists "cattle-csp-billing-adapter-system"
+        helm upgrade --install csp-adapter-crd /cnab/app/charts/csp-adapter-crd/100.0.0+up0.1.0 --namespace cattle-csp-billing-adapter-system --atomic --timeout ${helm_timeout}
+        echo "CSP Rancher Usage Operator CRD successfully installed"
+        echo "Installing CSP Rancher Usage Operator"
+	EnsureKubernetesNamespaceExists "cattle-csp-billing-adapter-system"
+	CreateRancherCACertificateSecrets
+        helm upgrade --install csp-rancher-usage-operator /cnab/app/charts/csp-rancher-usage-operator/100.0.0+up0.1.0 ${rancher_usage_operator_image_arguments} --namespace cattle-csp-billing-adapter-system --atomic --timeout ${helm_timeout}
+        echo "CSP Rancher Usage Operator successfully installed"
+        echo "Installing Rancher CSP Billing Adapter"
+	EnsureKubernetesNamespaceExists "cattle-csp-billing-adapter-system"
+        helm upgrade --install rancher-csp-billing-adapter /cnab/app/charts/rancher-csp-billing-adapter/100.0.0+up0.1.0 ${rancher_billing_adapter_image_arguments} ${azure_billing_arguments} ${aws_billing_arguments} --namespace cattle-csp-billing-adapter-system --atomic --timeout ${helm_timeout}
```

Comment:
should we add some checks that make sure the pods are in Running state?

Reply:
I think --atomic will do the checking.